backend/src/main/java/ca/bc/gov/app/repository/fsa/ClientTypeCodeRepository.java

```diff
@@ -0,0 +1,17 @@
+package ca.bc.gov.app.repository.fsa;
+
+import ca.bc.gov.app.entity.fsa.ClientTypeCodeEntity;
+import java.time.LocalDate;
+import org.springframework.data.r2dbc.repository.Query;
+import org.springframework.data.repository.reactive.ReactiveCrudRepository;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Flux;
+
+@Repository
+public interface ClientTypeCodeRepository
+    extends ReactiveCrudRepository<ClientTypeCodeEntity, String> {
+
+  @Query("SELECT * FROM nrfc.client_type_code WHERE (expiry_date is null or expiry_date > :activeDate) and effective_date <= :activeDate")
```

Comment:
any specific reason to write the query, is not it auto-generated by spring r2dbc from the method name?

Reply:
Not for this special scenario. We can do and, or, less than equal but mixing some of them together will generate a different result. In this scenario we need an or alongside an and. Besides, the method name would end up being grotesque.